062-support-tiered-storage.md

```diff
@@ -0,0 +1,169 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for RemoteStorageManager and RemoteLogMetadataManager, users are able to adopt tiered storage feature on Strimzi easily.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+### RemoteStorage
+
+Add a `tieredStorage` config under KafkaClusterSpec, which would support configuration for all tiered storage related configuration.
+
+The CustomTieredStorage class contains configuration for various aspects, including RLM config, RSM specific config, RLMM specific config. 
```

Comment:
It's the first time you mention a `CustomTieredStorage` class.
I can imagine it's the class implementing the `tieredStorage` field in the Strimzi API related package? The class from the corresponding CRD part is generated.
If yes, it's not really understandable for a reader who doesn't know how Strimzi API model is designed and how the corresponding CRD is generated. I would avoid to talk about the class but still sticking with a custom resource field. Mentioning the structure of the class later if really needed.

Also being clear about what RLM, RSM, RLMM are.

Reply:
Updated.